Patch that fixes several problems for building on amd64. Thanks Fedora guys.
=====================================================================
--- freeimage.orig/Source/FreeImage/PluginBMP.cpp
+++ freeimage/Source/FreeImage/PluginBMP.cpp
@@ -192,6 +192,8 @@ LoadPixelData(FreeImageIO *io, fi_handle
 	return TRUE;
 }
 
+#define MIN(a, b) ( ((a) < (b))? (a):(b) )
+
 /**
 Load image pixels for 4-bit RLE compressed dib
 @param io FreeImage IO
--- freeimage.orig/Source/FreeImage.h
+++ freeimage/Source/FreeImage.h
@@ -141,8 +141,10 @@ typedef uint8_t uint8_t;
 typedef uint16_t uint16_t;
 typedef uint32_t uint32_t;
 typedef int32_t int32_t;
+#ifndef _LIBRAW_TYPES_H
 typedef int64_t int64_t;
 typedef uint64_t uint64_t;
+#endif
 #else
 // MS is not C99 ISO compliant
 typedef long BOOL;
